packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -1054,9 +1054,35 @@ class ReactDOMServerRenderer {
             return '';
           }
           case REACT_CONTEXT_TYPE: {
-            const consumer: ReactConsumer<any> = (nextChild: any);
-            const nextProps: any = consumer.props;
-            const nextValue = consumer.type._currentValue;
+            let reactContext: ReactContext<any> = (nextChild: any).type;
```

Comment:
Seems a little odd that we're any-casting something that explicitly _might not be_ `ReactContext` to that type for DEV mode.